cmd/baton-calendly/main.go

```diff
@@ -5,40 +5,49 @@ import (
 	"fmt"
 	"os"
 
-	"github.com/conductorone/baton-sdk/pkg/cli"
+	"github.com/conductorone/baton-calendly/pkg/connector"
+	configSchema "github.com/conductorone/baton-sdk/pkg/config"
 	"github.com/conductorone/baton-sdk/pkg/connectorbuilder"
+	"github.com/conductorone/baton-sdk/pkg/field"
 	"github.com/conductorone/baton-sdk/pkg/types"
 	"github.com/grpc-ecosystem/go-grpc-middleware/logging/zap/ctxzap"
+	"github.com/spf13/viper"
 	"go.uber.org/zap"
+)
 
-	"github.com/conductorone/baton-calendly/pkg/connector"
+const (
+	version       = "dev"
+	connectorName = "baton-calendly"
 )
 
-var version = "dev"
+var (
+	Token               = field.StringField(connector.Token, field.WithRequired(true), field.WithDescription("Personal Access Token used to authenticate with the Calendly API."))
+	configurationFields = []field.SchemaField{Token}
+)
 
 func main() {
 	ctx := context.Background()
-
-	cfg := &config{}
-	cmd, err := cli.NewCmd(ctx, "baton-calendly", cfg, validateConfig, getConnector)
+	_, cmd, err := configSchema.DefineConfiguration(ctx,
+		connectorName,
+		getConnector,
+		field.NewConfiguration(configurationFields),
+	)
 	if err != nil {
 		fmt.Fprintln(os.Stderr, err.Error())
 		os.Exit(1)
 	}
 
 	cmd.Version = version
-	cmdFlags(cmd)
-
 	err = cmd.Execute()
 	if err != nil {
 		fmt.Fprintln(os.Stderr, err.Error())
 		os.Exit(1)
 	}
 }
 
-func getConnector(ctx context.Context, cfg *config) (types.ConnectorServer, error) {
+func getConnector(ctx context.Context, cfg *viper.Viper) (types.ConnectorServer, error) {
 	l := ctxzap.Extract(ctx)
-	cb, err := connector.New(ctx, cfg.Token)
+	cb, err := connector.New(ctx, cfg)
```

Comment:
I think you still need to call
```go
if err := validateConfig(ctx, cfg); err != nil {
  return nil, err
}
```
to validate that `token` is required.

Reply:
Yes, token is required and because of that, We do not hit the validateConfig, the only way for hitting the validateConfig is when the field is not required. Currently, we get a message like this "required flag(s) "token" not set"